wheel/metadata.py

```diff
@@ -47,20 +47,27 @@ def generate_requirements(extras_require):
     extras_require is a dictionary of {extra: [requirements]} as passed to setup(),
     using the empty extra {'': [requirements]} to hold install_requires.
     """
+    seen = set()
     for extra, depends in extras_require.items():
         condition = ''
         if extra and ':' in extra:  # setuptools extra:condition syntax
             extra, condition = extra.split(':', 1)
             extra = pkg_resources.safe_extra(extra)
         if extra:
-            yield ('Provides-Extra', extra)
+            tup = ('Provides-Extra', extra)
```

Comment:
Why not collect all the Provides-Extra in a set() and yield them all at the end?

Reply:
That seems just a different way of doing the same thing.

Reply:
It would eliminate the extra 'if' statements.